### PR TITLE
ARROW-8126: [C++][Compute] Add nth-to-indices kernel benchmark

### PR DESCRIPTION
**Pull request by cyb70289**

### CODE REVIEW

File: cpp/src/arrow/compute/kernels/nth_to_indices_benchmark.cc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+static void NthToIndicesBenchmark(benchmark::State& state,
+                                  const std::shared_ptr<Array>& values, int64_t n) {
+  FunctionContext ctx;
+  for (auto _ : state) {
+    std::shared_ptr<Array> out;
+    ABORT_NOT_OK(NthToIndices(&ctx, *values, n, &out));
+    benchmark::DoNotOptimize(out);
+  }
+}
```
Comment (pitrou) [inline cpp/src/arrow/compute/kernels/nth_to_indices_benchmark.cc]: Can you perhaps call `state.SetItemsProcessed` at the end of this function? Also the same for `sort_to_indices_benchmark.cc`

**Review by pitrou — CHANGES_REQUESTED** [review]
Thank you for doing this. Just a minor improvement request.

File: cpp/src/arrow/compute/kernels/nth_to_indices_benchmark.cc (hunk 1 above)
Comment (cyb70289) [reply cpp/src/arrow/compute/kernels/nth_to_indices_benchmark.cc]: Done

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
Thank you very much @cyb70289 . Looking forward to more contributions :-)

**Pull request closed without merging**